File: ASP.NET Core + Angular/Task/OnlineBookStoreAPI/Helpers/OrderParams.cs
using OnlineBookStoreAPI.Constants;

namespace OnlineBookStoreAPI.Helpers
{
    public class OrderParams : PaginationParams
    {
        public string SortBy { get; set; } = Const.Date;
        public string SortOrder { get; set; } = Const.DESCENDING;
    }
}

File: ASP.NET Core + Angular/Task/client/src/app/_models/orderParams.ts
import { Constants } from "./constants";

export class OrderParams {
    pageNumber: number;
    pageSize: number;
    sortBy: string;
    sortOrder: string;

    constructor(sortBy?: string, sortOrder?:string, pageNumber?: number, pageSize?:number) {
        this.pageNumber = pageNumber || Constants.pageNumber;
        this.pageSize = pageSize || Constants.pageSize;
        this.sortBy = sortBy || Constants.sortByDate;
        this.sortOrder = sortOrder || Constants.sortOrderDsc;
    }
}